src/main/kotlin/blackjack/domain/CardPack.kt

```diff
@@ -0,0 +1,32 @@
+package blackjack.domain
+
+import blackjack.enums.CardShape
+import blackjack.enums.CardType
+
+class CardPack {
+    private val cards = mutableListOf<Card>()
+
+    init {
+        CardShape.values().forEach { cardShape ->
+            createCardsOf(cardShape)
+        }
+        cards.shuffle()
+    }
+
+    fun pickCard(): Card {
+
+        if (cards.isEmpty()) {
+            throw IllegalStateException("모든 카드가 사용되었습니다.")
+        }
```

Comment:
`check()`는 값을 만족하지 않으면 `IllegalStateException`을 발생시킵니다.

```suggestion
        check(cards.isNotEmpty()) { "모든 카드가 사용되었습니다." }
```